databricks/koalas/datetime.py

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright (C) 2019 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Date/Time related functions on Koalas Series
+"""
+
+import functools
+from pyspark.sql.types import DateType, TimestampType, LongType, StringType
+
+from databricks.koalas.series import Series
+from databricks.koalas.utils import lazy_property
+
+import pyspark.sql.functions as F
+
+
+def defer_to_pandas(output_type):
+    """Wraps a function that operates on pd.Series to work with DatetimeMethods.
+
+    The origin function operates on pd.Series and the wrapped function operates
+    on DatetimeMethods.
+    """
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(self, *args):
+            scol = F.pandas_udf(
+                lambda series: func(series, *args),
+                output_type
+            )(self._data._scol)
+            return Series(
+                scol, anchor=self._data._kdf, index=self._data._index_info)
+        return wrapper
+    return decorator
+
+
+def defer_to_spark(output_type):
+    """Wraps a function that operates on spark.sql.Column to work with DatetimeMethods.
+
+    The origin function operates on spark.sql.Column and the wrapped function
+    operates on DatetimeMethods.
+    """
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(self, *args):
+            scol = func(self._data._scol, *args).cast(output_type)
+            return Series(scol, anchor=self._data._kdf, index=self._data._index_info)
+        return wrapper
+    return decorator
+
+
+class DatetimeMethods:
+    """Date/Time methods for Koalas Series"""
+    def __init__(self, series):
+        if not isinstance(series.spark_type, (DateType, TimestampType)):
+            raise ValueError(
+                "Cannot call DatetimeMethods on type {}"
+                .format(series.spark_type))
+        self._data = series
+
+    # Properties
+    @lazy_property
```

Comment:
why do you need lazy property? The performance difference is negligible.

Reply:
I meant compared to regular `@property`

Reply:
I am not too sure about using property here. When user creates a koalas Series, I think it's a little too much to also create lots of accessor series along with it. It increases the number of Series we create by a lot I think.